### DISCUSSION TITLE
Retrieve data from portal form

### DISCUSSION THREAD

**Discussion by Herobyte** [khoih-prog/WiFiManager_NINA_Lite#28] [Ideas]
Hi!
I need retrieve data from the form, example the Blynk_Server1 and put it in a variable, is possible?

`#if USE_DYNAMIC_PARAMETERS

  #define MAX_BLYNK_SERVER_LEN      34
  #define MAX_BLYNK_TOKEN_LEN       34
  
  char Blynk_Server1 [MAX_BLYNK_SERVER_LEN + 1]  = "account.duckdns.org";
  char Blynk_Token1  [MAX_BLYNK_TOKEN_LEN + 1]   = "token1";
  
  char Blynk_Server2 [MAX_BLYNK_SERVER_LEN + 1]  = "account.ddns.net";
  char Blynk_Token2  [MAX_BLYNK_TOKEN_LEN + 1]   = "token2";
  
  #define MAX_BLYNK_PORT_LEN        6
  char Blynk_Port   [MAX_BLYNK_PORT_LEN + 1]  = "8080";
  
  #define MAX_MQTT_SERVER_LEN      34
  char MQTT_Server  [MAX_MQTT_SERVER_LEN + 1]   = "mqtt.duckdns.org";
  
  MenuItem myMenuItems [] =
  {
    { "sv1", "Blynk Server1", Blynk_Server1,  MAX_BLYNK_SERVER_LEN },
    { "tk1", "Token1",        Blynk_Token1,   MAX_BLYNK_TOKEN_LEN },
    { "sv2", "Blynk Server2", Blynk_Server2,  MAX_BLYNK_SERVER_LEN },
    { "tk2", "Token2",        Blynk_Token2,   MAX_BLYNK_TOKEN_LEN },
    { "prt", "Port",          Blynk_Port,     MAX_BLYNK_PORT_LEN },
    { "mqt", "MQTT Server",   MQTT_Server,    MAX_MQTT_SERVER_LEN },
  };`

This is the code, if I need retrieve the Blynk_Server1 put in the form and make this in variable, how make?
Or make another field and retrieve the data

Thanks

**Comment by khoih-prog** [comment]
Have a look at 

https://github.com/khoih-prog/WiFiManager_NINA_Lite/blob/cdb027a21b68ce4cc2464948a60ecd74b17ed8c1/examples/SAMD_WiFiNINA/SAMD_WiFiNINA.ino#L101-L111

then you'll see, for example Blynk_Server1 is index 0 in the myMenuItems, then you can use the null-terminated string at

```
myMenuItems[0].pdata
```

and so on

**Reply by Herobyte** [reply]
I try, but not understand.. I have 2 define, how retrieve this 2 define from the form ?

**Reply by khoih-prog** [reply]
You have to read a little bit more about pointer, etc.

Just a hint without explanation. Try, for example,

```
char *  Blynk_Server1 = myMenuItems[0].pdata;
```

then use it normally, such as

```
Serial.println(Blynk_Server1);
```

**Reply by Herobyte** [reply]
Sorry, but i not are espert in arduino programming..

I use a telegram library and I have this:

UniversalTelegramBot bot(BOT_TOKEN, client);
In this moment the BOT_TOKEN is picked up from a #define (#define BOT_TOKEN "TOKEN")

How picked up the Token from form portal?.. Ex: from Blynk_Server1 field ?

Thanks

**Reply by khoih-prog** [reply]
Sorry I don't have time to help everybody here. I'd rather spend time to help many mode people with libraries than devoting time to help just one person.

If not OK, I suggest you post for help on Arduino, Telegram, etc. Forum. Remember to post full code or your request will be ignored.

Good Luck,

**Reply by khoih-prog** [reply]
I saw you posted something on Arduino Forum, and that's a good sign that you are not passive and waiting only.

So the simple solution can be 

```
char *myBOT_TOKEN =  myMenuItems[0].pdata;

UniversalTelegramBot bot(myBOT_TOKEN, client);
```

**Reply by Herobyte** [reply]
Thank you, you save me life :D
